shavee-bin/src/args.rs

```diff
@@ -170,24 +162,42 @@ impl Sargs {
                 // If PAM mode is enabled, then the user name from PAM_USER
                 // environment variable is added to the end of ZFS dataset
                 // before it is mounted.
-                // Clap has already validated PAM_USER existence.
-                dataset.push('/');
-                dataset.push_str(env::var("PAM_USER").expect(UNREACHABLE_CODE).as_str());
-                String::from("pam")
+                let user = match env::var("PAM_USER") {
+                    Ok(u) => u,
+                    Err(e) => {
+                        // FIX for "[bug] Command line parse error #7"
+                        // If PAM mode is selected but PAM_USER env doesn't exists
+                        // then return Err(error message)
+                        let error_message = e.to_string().to_owned();
+                        return Err(
+                            clap::Error::with_description(
+                                &error_message[..], clap::ErrorKind::EmptyValue))
+                        },
+                    };
+                // append the value from "PAM_USER" to the end of dataset, if it exists
+                // otherwise keep None
+                dataset = dataset
+                    .map(|d| {
```

Comment:
Like the comment on line 138, here is an example using `map`.

Remember that this code is `closure` and [*captures*](https://doc.rust-lang.org/rust-by-example/fn/closures/capture.html) the environment variables.